src/_shared/js/capi-images.js

```diff
@@ -0,0 +1,65 @@
+// Creates a set of source elements for an image.
+function buildSources (sourceData) {
+
+	let sourcesFragment = document.createDocumentFragment();
+
+	return sourceData.reduce((sources, source) => {
```

Comment:
Yes, there's no real reason to use a reducer with a DOM object. If you want to avoid access to the closure, you can pass `sourcesFragment` as the context in `forEach`, like that `sourceData.forEach(fn, sourcesFragment)` and inside `fn` you must reference `sourcesFragment` as `this`. This is by the way a performance tip that is used and abused in bluebird.